include/mppic/optimizers/xtensor/critic_function.hpp

```diff
@@ -37,26 +32,39 @@ class CriticFunction
     costmap_ = costmap_ros_->getCostmap();
     parameters_handler_ = param_handler;
 
+    auto getParam = parameters_handler_->getParamGetter(name_);
+    getParam(enabled_, "enabled", true);
+
     initialize();
   }
 
+  void score(models::CriticFunctionData & data)
+  {
+    if (data.stop_flag) {
```

Comment:
Can you explain the stop flag bit?

Reply:
If all trajectories collide, critic manager and optimzer got this signal, stop evaluating critics and reset optimizer

Reply:
Oh great move :+1: Would that require the obstacle critic is run first?

Reply:
I don't know if this would ruin some of the tensor representation's benefits, but would it be possible to process only trajectories that are valid so that this stop_flag is processed on a per-trajectory basis? That way we can skip the collision trajectories and only process the valid ones.

Reply:
> Oh great move +1 Would that require the obstacle critic is run first? How often is it the case that all trajectories fail in that way? I would think that would be rare.

With previous settings it was actually very often, and this flag helps to determine this. 
With current settings in this branch rarely


> I don't know if this would ruin some of the tensor representation's benefits, but would it be possible to process only trajectories that are valid so that this stop_flag is processed on a per-trajectory basis? That way we can skip the collision trajectories and only process the valid ones. In DWB we do this, but we don't use tensor representations / libraries that under the hood might have runtime improvements based on that structure

It is possible, and i think it's nice idea to try out. For now we just give them huge cost, so their controls almost no effect on final controls